spec/integration/downloading_documents.rb

```diff
@@ -0,0 +1,46 @@
+require 'rails_helper'
+
+feature 'Downloading documents' do
+  let(:notice) { create(:dmca, :with_pdf, :with_image, :with_document) }
+  let(:token_url) { create(:token_url, notice: notice, documents_notification: true) }
+
+  scenario 'will redirect to the root page when a user has no access to download a document', js: true do
+    visit notice.file_uploads.first.url
+
+    expect(page).to have_current_path(root_path)
+    expect(page).to have_content('You are not allowed to download this document.')
+  end
+
+  scenario 'will download a document for a user with access', js: true do
+    sign_in(create(:user, :admin))
```

Comment:
Do other user classes, like researchers, have access? If so we should test that also.

Reply:
Done

Reply:
👍